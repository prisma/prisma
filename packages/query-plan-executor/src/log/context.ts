import { context } from '@opentelemetry/api'

import { Logger } from './logger.ts'

const LOG_CONTEXT = Symbol('LogContext')

/**
 * Gets the logger from the current OpenTelemetry context.
 *
 * @throws {Error} if no logger is found in the context or if the value is not a {@link Logger}
 */
export function getActiveLogger(): Logger {
  const logger = context.active().getValue(LOG_CONTEXT)

  if (!logger) {
    throw new Error('No active logger found')
  }

  if (logger instanceof Logger) {
    return logger
  }

  throw new Error('Active logger is not an instance of Logger')
}

/**
 * Runs a function with a new OpenTelemetry context containing the provided logger.
 *
 * @param logger The logger to make active
 * @param fn The function to run with the logger in context
 * @returns The return value of the function
 */
export function withActiveLogger<T>(logger: Logger, fn: () => T): T {
  return context.with(context.active().setValue(LOG_CONTEXT, logger), fn)
}
