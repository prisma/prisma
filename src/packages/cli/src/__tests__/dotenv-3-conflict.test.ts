import { consoleContext, Context } from './__helpers__/context'

const ctx = Context.new().add(consoleContext()).assemble()

it('should throw error', async () => {
  ctx.fixture('dotenv-3-conflict')
  process.argv.push('--version')
  try {
    await import('../bin')
  } catch (e) {
    expect(e).toMatchSnapshot()
  }
})
