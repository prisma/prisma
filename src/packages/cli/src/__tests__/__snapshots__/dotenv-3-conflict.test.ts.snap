// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should throw error 1`] = `

      You are trying to load duplicate env variables which are already present in your project root .env
      	root env path: /tmp/dir/.env
      	schema env path: ./prisma/.env
      	Duplicates:
      		SHOULD_THROW 
      
`;
